### ISSUE TITLE
How to load saved model?

### ISSUE THREAD

**Issue by zohaibjan** [taipingeric/yolo-v4-tf.keras#19]
How do I load saved model after custom training ?

**Comment by bhorsfield** [comment]
@zohaibjan, did you ever find a solution to this problem?

**Comment by zohaibjan** [comment]
@bhorsfield sorry mate I moved away from this and went to use Darknet Yolov4 by Alexyab. I did manage to get it partly working but that required a lot of effort as layers needed to be unwrapped so just left it as is. If you are looking for simpler implementation then I will suggest Nvidia TAO Toolkit

**Comment by bhorsfield** [comment]
@zohaibjan, thanks for the tip, I've never heard of the TAO Toolkit before.  It looks like an excellent platform, especially for less experienced ML developers.  I will definitely check it out.

**Comment by bhorsfield** [comment]
For anyone who is still trying to use the YOLOv4 model in this GitHub repo, I have some progress to report:

In order to load a saved model after custom training (i.e. with your own dataset), the model **must** have been saved using the `save_model()` method in the **Yolov4** class.

Unfortunately, this means that you can't load a model that was saved using the **ModelCheckpoint** callback, due to the fact that the callback will save `self.training_model`, which is the _wrong model_.  (The model we want to save is actually `self.yolo_model`.)  
 
If, like me, you **really** want to use the **ModelCheckpoint** callback to automatically save the model with the lowest validation loss during training, you will have to create a custom version of this callback that saves `self.yolo_model` instead of `self.training_model`.

Note that @Klimroth came to a similar conclusion in another discussion thread:  [https://github.com/taipingeric/yolo-v4-tf.keras/issues/18#issue-1129563144](url).  However, no solution was ever reported.

**Comment by bhorsfield** [comment]
I have created a custom callback that will enable you to save a checkpoint during training in a format that can be loaded back in from file later using the `load_model()` method.  Refer to Issue https://github.com/taipingeric/yolo-v4-tf.keras/issues/18#issuecomment-1455405090 for details & source code.
